contracts/example/Colors.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.11;
+
+contract Colors {
+    bytes32[] public colorsArray;
+
+    uint256 public colorCounter = 0;
+
+    event setColorEvent(bytes32 color);
+    event metadataDepositorEvent(address depositorAddress);
+
+    function setColorOnDeploy(bytes32 color) public {
+      colorsArray.push(color);
+    }
+
+    function setColor(bytes32 metadataDepositor, bytes32 color) public {
```

Comment:
Not them, but this can't be an address because it is not necessarily Ethereum address, so if it is an 32bytes address of lets say Substrate you can't typecast it.